### PR TITLE
EMCal CorrFW: Implement cluster energy scaling/smearing component

### PR DESCRIPTION
**Pull request by jdmulligan**
Please do not approve yet.

This is a draft of how we could implement scaling/smearing of the cluster energy in order to evaluate systematics, e.g. in jet analyses, but also available for everyone.

@efranzis @raymondEhlers please let me know what you think.

### CODE REVIEW

**Comment by jdmulligan** [comment]
+test

File: PWG/EMCAL/EMCALtasks/AliEmcalCorrectionClusterEnergyShift.cxx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      
+      // Smear energy (if applicable)
+      if (fSmearingWidth > 1e-6) {
+        energyclus *= (1 + fRandom.Gaus(0, fSmearingWidth) );
+      }
+      
+      // Set the shifted energy to the user-defined energy field
+      cluster = static_cast<AliVCluster *>(clusIterator->second);
+      if (cluster->IsEMCAL() && energyclus > 0) {
+        cluster->SetUserDefEnergy(AliVCluster::kUserDefEnergy1, energyclus);
```
Comment (raymondEhlers) [inline PWG/EMCAL/EMCALtasks/AliEmcalCorrectionClusterEnergyShift.cxx]: Per our conversation, I think this is a scaling factor

Comment (raymondEhlers) [reply PWG/EMCAL/EMCALtasks/AliEmcalCorrectionClusterEnergyShift.cxx]: Never mind, this is fine. In [`AliVCluster`](https://github.com/alisw/AliRoot/blob/f7eab9024d7187361500944ca662e0f1c9c57333/STEER/AOD/AliAODCaloCluster.h#L136), it is set as:

```c++
  void        SetUserDefEnergy(VCluUserDefEnergy_t t, Double_t e)               { fUserDefEnergy[t] = E() > 1e-6 ? e / E() : 1.  ; }
```

**Comment by efranzis** [comment]
Looks good to me. I just wonder if the name ..CorrectionClusterEnergyShift is the best description since it allows a smearing too. What about something in the direction of ..CorrectionArtificialClusterEnergyChange or CorrectionClusterEnergyVariation or CorrectionClusterEnergyModification ... EnergyDistortion etc

**Comment by jdmulligan** [comment]
Hi @efranzis, Okay I will change the change to EnergyVariation.

**Comment by jdmulligan** [comment]
+test

**Comment by jdmulligan** [comment]
+1

**Pull request merged**